core/pkg/server/sender.go

```diff
@@ -312,7 +313,8 @@ func (s *Sender) sendRequest(record *service.Record, request *service.Request) {
 	case *service.Request_Defer:
 		s.sendDefer(x.Defer)
 	case *service.Request_LogArtifact:
-		s.sendLogArtifact(record, x.LogArtifact)
+		msg := x.LogArtifact
+		s.sendLogArtifact(record, msg.Artifact, msg.HistoryStep, msg.StagingDir)
```

Comment:
why do you need to pass all these explicitly? i think you can just pass the x.LogArtifact and follow the same pattern as the other functions?

Reply:
oh i see... you want to reuse the LogArtifact function, makes sense. 
What do you think about:

```
case *service.Record_Artifact:
     s.sendArtifact(record, x.Artifact)
case *service.Request_LogArtifact:
     s.sendLogArtifact(record, x.Artifact)

func s.sendArtifact(record, x.Artifact):
      logArtifact(...)

func s.sendLogArtifact(record, x.LogArtifact):
      logArtifact(...)

func logArtifact( ...) << implements the shared logic
```

Reply:
Made some changes. Initially, I made the case where record is of type `Record_Artifact` also call `sendLogArtifact()`. Instead, made a new method for `sendArtifact()` now